nexus/db-queries/src/db/pool.rs

```diff
@@ -173,18 +170,88 @@ impl Pool {
                 err.into_inner()
             }
         };
+        Self::new_common(inner, log.clone())
+    }
+
+    fn new_common(
+        inner: qorb::pool::Pool<AsyncConnection>,
+        log: Logger,
+    ) -> Self {
+        let (quiesce, _) = watch::channel(Quiesce {
+            new_claims_allowed: ClaimsAllowed::Allowed,
+            claims_held: IdOrdMap::new(),
+        });
         Pool {
+            next_id: AtomicU64::new(0),
             inner,
-            log: log.clone(),
+            log,
             terminated: std::sync::atomic::AtomicBool::new(false),
+            quiesce,
         }
     }
 
     /// Returns a connection from the pool
-    pub async fn claim(
-        &self,
-    ) -> anyhow::Result<qorb::claim::Handle<QorbConnection>> {
-        Ok(self.inner.claim().await?)
+    pub async fn claim(&self) -> Result<DataStoreConnection, Error> {
+        let id = self.next_id.fetch_add(1, Ordering::SeqCst);
+        let held_since = Utc::now();
+        let debug = Backtrace::force_capture().to_string();
+        let allowed = self.quiesce.send_if_modified(|q| {
+            if let ClaimsAllowed::Disallowed = q.new_claims_allowed {
+                false
+            } else {
+                q.claims_held
+                    .insert_unique(HeldDbClaimInfo { id, held_since, debug })
```

Comment:
this is probably fine, but to say it explicitly - this is a capture of where the claim *started*, not where the claim may be "stuck in-usage", right?